content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx

```diff
@@ -0,0 +1,170 @@
+- `result` extensions (for future release): [add custom results objects to Prisma Client](/concepts/components/prisma-client/client-extensions/client-extensions-result)
+- `query` extensions (for future release): [create custom Prisma Client queries](/concepts/components/prisma-client/client-extensions/client-extensions-query)
+
+</TopBlock>
+
+## About Prisma Client extensions
+
+To create Prisma Client extensions, use the `$extends` [client-level method](/reference/api-reference/prisma-client-reference#client-methods), and the appropriate extension type (`model`, `client`, `query` or `result`).
+
+You can use [multiple independent extensions at the same time](#multiple-extensions), each in is own isolated instance of Prisma Client.
+
+You can also [share your Prisma Client extensions](#share-extensions) with other Prisma users, and [import Prisma Client extensions developed by other users](#install-a-packaged-extension) into your Prisma project.
+
+## Prerequisites
+
+To use Prisma Client extensions, you must enable the `clientExtensions` feature flag in the generator block of your `schema.prisma` file, as follows:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["clientExtensions"]
+}
+```
+
+## Multiple extensions
+
+Each time that you call `$extends`, you create an _extended client_. This extended client is a forked state of the standard Prisma Client. The original client does not change.
+
+Every extended client operates independently, in an isolated instance. It cannot conflict with other extended clients or the original client.
+
+You can also combine two or more extensions into one extension, if you want the functionality from both extensions to apply in the same client instance.
+
+### Examples of isolated client instances
+
+Example uses of isolated client instances include the following:
+
+- Implement row-level security (RLS), where each http request has its own client with its own RLS extension, customized with session data. This can keep each user entirely separate, each in a separate client.
+- Add a `user.current()` method for the `user` model to get the current logged-in user
+- Enable more verbose logging for requests if a debug cookie is set
+- Attach a unique request id to all logs so you can correlate them later
+- Remove a `delete` method from models unless the application calls the admin endpoint and the user has the necessary priviliges
+
+### Declare multiple extensions
+
+The following example declares two separate extensions and one combined extension:
+
+```ts
+// First of all, store your original prisma client in a variable as usual
+const prisma = new PrismaClient()
+
+// Declare extension A
+const extendsA = prisma.$extends(extensionA)
+
+// Declare extension B
+const extendsB = prisma.$extends(extensionB)
+
+// Declare an extension that is a combination of A and B
+const extendsAB = prisma.$extends(extensionA).$extends(extensionB)
```

Comment:
```suggestion
// Declare an extension that is a combination of A and B
const extendsAB = prisma.$extends(extensionA).$extends(extensionB)

// Declare an extension that is a stacking combination of A and B
const extendsAplusB = extensionA.$extends(extensionB)
```
Does this work as well?

Reply:
@janpio  What is the benefit of the "stacked" declaration versus the "combination" declaration?

Reply:
I'll ask in the channel about whether this is possible